### PR TITLE
Fix some issues reported by LGTM

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

File: themes/bootstrap3/js/check_save_statuses.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ function runSaveAjaxForQueue() {
           displaySaveStatus(response.data.statuses[id], saveStatusEls[id]);
 
           // Remove populated ids from the queue
-          for (var j = 0; j < saveStatusObjs; j++) {
+          for (var j = 0; j < saveStatusObjs.length(); j++) {
```
Comment (EreMaijala) [inline themes/bootstrap3/js/check_save_statuses.js]: length is a property, not a method, right?

**Comment by stweil** [comment]
See also https://lgtm.com/projects/g/vufind-org/vufind/alerts for more alerts.

LGTM also supports [CI with GitHub](https://lgtm.com/projects/g/vufind-org/vufind/ci/).

**Comment by demiankatz** [comment]
Thanks, @stweil -- looks like Travis is now failing because there's a missing `var` in the creation of the `init` function in `map_tab_leaflet`. I think the `storage` check in `embedded_record.js` may be intended as a verification of browser capabilities, but @crhallberg can probably clarify whether that still serves any purpose. I also think we should evaluate what that broken loop is actually supposed to be doing in `check_save_statuses.js` -- if we haven't noticed any problems related to the bug you fixed, I wonder if we need a better test or if the code is superfluous. In any case, I'd want to look carefully at a loop that uses `.length` as the end condition but uses `splice` to change the length of the array inside the loop. I have a feeling that may still not be right!

**Comment by stweil** [comment]
Regarding the `storage` check: it duplicates existing code, so I don't think it verifies browser capabilities. And yes, the broken loop needs a closer look by someone who knows the code better than me.

**Comment by demiankatz** [comment]
Thanks, @stweil, you're right about the `storage` check; I didn't look at enough of the context. I'll take a look at the broken loop when time permits (unless @crhallberg gets there first).

**Comment by demiankatz** [comment]
@stweil, I've taken care of the broken loop review -- that code was indeed completely incorrect in multiple ways, so save status information was being loaded repeatedly for no reason. The bug is now fixed. Thanks again for helping to track it down (along with these other lesser issues). I'll take a look at whether it makes sense to incorporate LGTM into our workflow going forward.

**Comment by demiankatz** [comment]
@stweil, I've enabled LGTM pull request integration -- it seems like it should provide useful information without impacting our workflows. I have also added the LGTM badges to our Jenkins instance so we can monitor them along with other CI stats:

https://vufind.org/jenkins/job/VuFind/

Thanks again for bringing this to our attention!

**Pull request merged**